### PR TITLE
Update CI and docs to remove references to default admin creds

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Starting in 2.12.0 release, security plugin is mandating an initial admin password to be set via env variable. This PR updates the docs and CI to accommodate this change.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by derek-ho** [comment]
https://github.com/opensearch-project/security/issues/3624

**Comment by derek-ho** [comment]
@dhrubo-os @greaa-aws can you folks take a look here and provide any feedback? This change will be happening for 2.12.0. Thanks!

File: docs/source/examples/online_retail_analysis.ipynb [hunk 1]
```diff
@@ -1294,7 +1294,7 @@
    "outputs": [
     {
      "data": {
```
Comment (derek-ho) [inline docs/source/examples/online_retail_analysis.ipynb:1296]: not sure what happened here, looks like new line diff?

**Comment by rawwar** [comment]
This change should have failed multiple tests as test client still has "admin" as password.  
https://github.com/opensearch-project/opensearch-py-ml/blob/5c93154bcfc04d8102d4d91294af36e02e0df802/tests/__init__.py#L38


I haven't looked into it properly. But, will take a look and respond.

**Comment by DarshitChanpura** [comment]
@dhrubo-os @greaa-aws @ylwu-amzn Can we get some reviews on this PR?

**Comment by dhrubo-os** [comment]
Could you please resolve the merge conflict?

**Comment by derek-ho** [comment]
> Could you please resolve the merge conflict?

Done

**Review by mingshl — APPROVED** [review]

**Review by dhrubo-os — APPROVED** [review]

**Comment by rawwar** [comment]
Merging this PR will require a issue in backlog so that we take care of updating passwords in the testcases and in other required places when we migrate to Opensearch>=2.12.0

**Comment by dhrubo-os** [comment]
@rawwar Thanks for the call out. Could you please create an issue for py-ml too? Thanks

**Comment by DarshitChanpura** [comment]
> @rawwar Thanks for the call out. Could you please create an issue for py-ml too? Thanks

Are you asking about #371?

**Comment by DarshitChanpura** [comment]
@greaa-aws @ylwu-amzn @b4sjoo @jngz-es @rbhavna Could we get 1 more review on this? Is there anything blocking from getting this merged? We are targeting 2.12 version release for this change.

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**